### DISCUSSION TITLE
errors for installation on a new computer, please help me!

### DISCUSSION THREAD

**Discussion by Wang-Yihu** [neams-th-coe/cardinal#484] [Q&A] [answered]
For installation of CARDINAL, I reinstall my Ubuntu 22.0. However, errors happen again and again. I suggest you can try install it in a new computer to summarize a perfect installation progress, ahahahahahahahahahahahah.╮(￣▽￣)╭╮(￣▽￣)╭╮(￣▽￣)╭╮(￣▽￣)╭
I follow the progress:
===============================================================================================
(The font looks so strange, I do not know what's going on……LOL)
https://cardinal.cels.anl.gov/start.html
Access
```
git clone https://github.com/neams-th-coe/cardinal.git
cd cardinal
```
Disable  NekRS
`export ENABLE_NEK=false`
Fetch Dependencies 
`./scripts/get-dependencies.sh`
```
Set these for .bashrc
export NEKRS_HOME=$HOME/cardinal/install
export HYPRE_DIR=$CONDA_PREFIX
export CC=mpicc
export CXX=mpicxx
export FC=mpif90
export OPENMC_CROSS_SECTIONS= MyCrossSection
```
===============================================================================================
Then I install the Conda MOOSE environment.
https://mooseframework.inl.gov/getting_started/installation/conda.html
I can make sure that for MOOSE Multiphysics, this can run without errors in .contrib/moose/test
```
conda activate moose
make -j4
```
===============================================================================================
Then I install OpenMC
https://docs.openmc.org/en/stable/quickinstall.html
I can make sure that for OpenMC, this can run without errors in .contrib/moose/test
```
conda activate openmc-env
./contrib/openmc/examples/pincell build_xml.py
openmc

```
===============================================================================================
I'm trying to` make `in the cardinal top direction, but it throw errors:
```
-- Found PkgConfig: /home/wangyihu/mambaforge3/envs/moose/bin/pkg-config (found version "0.29.2") 
-- Checking for module 'libmesh>=1.6.0'
--   Found libmesh, version 1.8.0-pre
-- Found ZLIB: /home/wangyihu/mambaforge3/envs/moose/lib/libz.so (found version "1.2.12") 
-- Found PNG: /home/wangyihu/mambaforge3/envs/moose/lib/libpng.so (found version "1.6.38") 
CMake Error at /home/wangyihu/mambaforge3/envs/moose/share/cmake-3.24/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find HDF5 (missing: HDF5_LIBRARIES HDF5_INCLUDE_DIRS
  HDF5_HL_LIBRARIES C HL) (found version "")
Call Stack (most recent call first):
  /home/wangyihu/mambaforge3/envs/moose/share/cmake-3.24/Modules/FindPackageHandleStandardArgs.cmake:594 (_FPHSA_FAILURE_MESSAGE)
  /home/wangyihu/mambaforge3/envs/moose/share/cmake-3.24/Modules/FindHDF5.cmake:1007 (find_package_handle_standard_args)
  CMakeLists.txt:113 (find_package)


-- Configuring incomplete, errors occurred!
See also "/media/wangyihu/bfc597c3-6afa-4175-a328-3c706bd907d7/wangyihu/cardinal/build/openmc/CMakeFiles/CMakeOutput.log".
```
===============================================================================================
I try to 
`sudo apt-get install libhdf5-dev`
but it does not work.
===============================================================================================
I try to install another OpenMC for another install method
https://anaconda.org/conda-forge/openmc
`conda install -c "conda-forge/label/cf202003" openmc`
Then I try `make` n the cardinal top direction, but it throw a different error
```
[  2%] Linking CXX static library ../../lib/libfmt.a
cd /media/wangyihu/bfc597c3-6afa-4175-a328-3c706bd907d7/wangyihu/cardinal/build/openmc/vendor/fmt && /home/wangyihu/mambaforge3/envs/moose/bin/cmake -P CMakeFiles/fmt.dir/cmake_clean_target.cmake
cd /media/wangyihu/bfc597c3-6afa-4175-a328-3c706bd907d7/wangyihu/cardinal/build/openmc/vendor/fmt && /home/wangyihu/mambaforge3/envs/moose/bin/cmake -E cmake_link_script CMakeFiles/fmt.dir/link.txt --verbose=1
CMAKE_AR-NOTFOUND qc ../../lib/libfmt.a CMakeFiles/fmt.dir/src/format.cc.o CMakeFiles/fmt.dir/src/os.cc.o
Error running link command: No such file or directory
make[3]: *** [vendor/fmt/CMakeFiles/fmt.dir/build.make:114: lib/libfmt.a] Error 2
make[3]: Leaving directory '/media/wangyihu/bfc597c3-6afa-4175-a328-3c706bd907d7/wangyihu/cardinal/build/openmc'
make[2]: *** [CMakeFiles/Makefile2:1067: vendor/fmt/CMakeFiles/fmt.dir/all] Error 2
make[2]: Leaving directory '/media/wangyihu/bfc597c3-6afa-4175-a328-3c706bd907d7/wangyihu/cardinal/build/openmc'
make[1]: *** [Makefile:124: all] Error 2
make[1]: Leaving directory '/media/wangyihu/bfc597c3-6afa-4175-a328-3c706bd907d7/wangyihu/cardinal/build/openmc'
```
===============================================================================================
Do you know how to figure this problem?
**Or do you know the perfect progress to install the cardinal. I do not mind reinstall Ubuntu again!**

**Comment by aprilnovak** [comment]
You don't need to separately install OpenMC or NekRS to use Cardinal. Cardinal will automatically build OpenMC and Nek for you, and I think this:

```
conda activate openmc-env
```

is causing a conflict because your system has 2 different versions of OpenMC. Can you please do

```
conda deactivate openmc-env
```

and then repeat the Cardinal build steps?

**Reply by Wang-Yihu** [reply]
Now I reinstall the Ubuntu. I repeat the CARDINAL install progress. I do not install openmc by myself, but it failed again.
The errors are the same as I've mentioned before.
The terminal throw errors:
```
CMake Error at /home/wangyihu/mambaforge3/envs/moose/share/cmake-3.24/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find HDF5 (missing: HDF5_LIBRARIES HDF5_INCLUDE_DIRS
  HDF5_HL_LIBRARIES C HL) (found version "")
Call Stack (most recent call first):
  /home/wangyihu/mambaforge3/envs/moose/share/cmake-3.24/Modules/FindPackageHandleStandardArgs.cmake:594 (_FPHSA_FAILURE_MESSAGE)
  /home/wangyihu/mambaforge3/envs/moose/share/cmake-3.24/Modules/FindHDF5.cmake:1007 (find_package_handle_standard_args)
  CMakeLists.txt:113 (find_package)
```
When I see the install progress again, I find the sentence
`If you are using the PETSc and libMesh that ship with MOOSE's conda environment, set PETSC_DIR and PETSC_ARCH to the PETSc install directory and architecture name and LIBMESH_DIR to the libMesh install directory. Then, skip directly to [Compile Cardinal ](https://cardinal.cels.anl.gov/start.html#compiling).`
Oops, I didn't do this. I ignore it.
**However, I do not know the meaning of _PETSc install directory and architecture name and LIBMESH_DIR to the libMesh install directory_**. I'm using PETSc and libMesh that ship with MOOSE's conda environment. There are many files and folders related the name _libmesh_ and _petsc_ in the direction of /mambaforge3/pkgs, /mambaforge3/envs/moose. I don't know how to set PETSC_DIR, PETSC_ARCH, LIBMESH_DIR.
Please tell me how to set them.
Thank you!

**Reply by aprilnovak** [reply]
Hi @Wang-Yihu - I've updated the Cardinal website to try and simplify the instructions a bit, in case it might be useful.

That error is telling you that Cardinal can't find HDF5. If you are using MOOSE's conda, you need to:

```
export HDF5_ROOT=$CONDA_PREFIX
```

**Reply by Wang-Yihu** [reply]
Ok, it does not work. After I put export `HDF5_ROOT=$CONDA_PREFIX` in .bashrc. I run `conda activate moose`; `make -j4`. It returns the same mistake.
```
CMake Error at /home/wangyihu/mambaforge3/envs/moose/share/cmake-3.24/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find HDF5 (missing: HDF5_LIBRARIES HDF5_INCLUDE_DIRS
  HDF5_HL_LIBRARIES C HL) (found version "")
Call Stack (most recent call first):
  /home/wangyihu/mambaforge3/envs/moose/share/cmake-3.24/Modules/FindPackageHandleStandardArgs.cmake:594 (_FPHSA_FAILURE_MESSAGE)
  /home/wangyihu/mambaforge3/envs/moose/share/cmake-3.24/Modules/FindHDF5.cmake:1007 (find_package_handle_standard_args)
  CMakeLists.txt:113 (find_package)
```
Maybe the conda environment from MOOSE in the current version does not have HDF5 you mentioned.
I try to 
```
cd $CONDA_PREFIX
ls
```
It returns
```
bin              envs     libexec      sbin   x86_64-conda_cos6-linux-gnu
compiler_compat  etc      LICENSE.txt  share  x86_64-conda-linux-gnu
condabin         include  man          shell
conda-meta       lib      pkgs         ssl
```
It seems that no folders related to HDF5.
Maybe we don't have a same MOOSE's conda...

**Reply by aprilnovak** [reply]
MOOSE's conda has HDF5. Whenever you change you environment (which means setting any environment variable), you usually need to delete any traces of your past attempts to build Cardinal. Can you try this? Sorry for not being more explicit in my last instructions.

```
cd cardinal
rm -rf build install
export HDF5_ROOT=$CONDA_PREFIX
make -j8
```

**Reply by Wang-Yihu** [reply]
So sad, it cannot work....Now, I‘ m buying a new computer...I'll get back soon and try to install again!(;´༎ຶД༎ຶ`)

**Reply by aprilnovak** [reply]
Hi @Wang-Yihu - I've tried making the build instructions even simpler - please check out this new page: https://cardinal.cels.anl.gov/start.html

If that still doesn't work, perhaps we can set up a time for a video chat to go over the steps together?

**Reply by Wang-Yihu** [reply]
Ok, thank you very much. I have sent email to anovak@anl.gov

**Reply by wangshuaiwu** [reply]
I also encountered this problem, I tried to use this method to solve it

**Reply by wangshuaiwu** [reply]
![image](https://user-images.githubusercontent.com/122420896/231925747-db8a7e6a-128e-4f77-8d12-b47cd8ac24fc.png)
The last problem was solved, and now this problem comes up again，I found a solution on the Internet, which is to execute this command
“sudo dnf install glibc-static”
But to no avail, the same problem still arises

**Comment by Wang-Yihu** [comment, marked as answer]
I have solved my problem.
https://github.com/neams-th-coe/cardinal/discussions/487
